otcextensions/sdk/nat/v2/_proxy.py

```diff
@@ -83,6 +83,25 @@ def update_gateway(self, gateway, **attrs):
         """
         return self._update(_gateway.Gateway, gateway, **attrs)
 
+    def find_gateway(self, name_or_id, ignore_missing=True):
+        """Find a single Nat Gateway
+
+        :param name_or_id: The name or ID of a zone
+        :param bool ignore_missing: When set to ``False``
+            :class:`~openstack.exceptions.ResourceNotFound` will be raised
+            when the gateway does not exist.
+            When set to ``True``, no exception will be set when attempting
+            to delete a nonexistent gateway.
+
+        :returns: ``None``
+        """
+        obj = self._find(_gateway.Gateway, name_or_id)
```

Comment:
you have here already "ignore_missing" param, so no need to reinvent checks and exceptions

Reply:
removed the extra checks/exception :)